### PR TITLE
Origin/search and pagination updates

### PR DESCRIPTION
**Pull request by m-ox**
# Change Type

* [X] Feature
* [X] Chore
* [ ] Bug Fix

# Change Level

* [X] major
* [ ] minor
* [ ] patch

# Further Information (screenshots, bug report links, etc.)

Aesthetic change
https://gyazo.com/ffe2235e489371198310b6544c479828

# Checklist

* [ ] Tested in supported environments (common browsers)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -46,30 +46,43 @@
...
       </p>
 
       <ul>
         <!-- filled from fetch -->
       </ul>
     </main>
 
+    <!-- quick scroll to top button -->
+    <script>
+      document.getElementById("scroll-top").onclick = function() {window.scrollTo(0, 0);}
```
Comment (zautumnz) [inline index.html]:
```suggestion
      document.getElementById("scroll-top").onclick = () => { window.scrollTo(0, 0) }
```

This project prefers ES2015+ syntax and no semicolons :)

File: script.js [hunk 2]
```diff
@@ -58,62 +59,87 @@
     return uniq([...pluralized, ...possibleAliases])
   }
 
-  const enrichResources = (resources) =>
+   enrichResources = (resources) => (
```
Comment (zautumnz) [inline script.js]:
```suggestion
   const enrichResources = (resources) => (
```

While it's technically legal to define without a declaration, it's not a great practice.

File: style.css [hunk 3]
```diff
@@ -96,5 +124,27 @@ small {
...
+  margin: 20px;
+  background-color: #cd6666;
+  color: #fff;
+  letter-spacing: 1px;
+  transition: 300ms;
 }
+
+#scroll-top:hover {
+  background-color: #b14848;
+}
```
Comment (zautumnz) [inline style.css:150]: Newline here please

File: style.css [hunk 4]
```diff
@@ -64,30 +67,55 @@ li:hover {
...
+  display: grid;
+  align-items: center;
+  gap: 10px;
+  grid-template-columns: 2fr 1fr;
+}
+
+.filtering button {
+  border-radius: 10rem;
+  border: 1px solid #e6af7c;
+  width: -webkit-fill-available;
```
Comment (zautumnz) [inline style.css:96]: Does this also need a `-moz-available` or something for FF support?

File: script.js [hunk 5]
```diff
@@ -58,62 +59,87 @@
...
-      }
-
-      // build the elements from the resources, and flip it since new
-      // links are added to the end of the yaml file
-      const items = enrichedResources.map(buildResourceItem).reverse()
-
-      // append all the items and add the filter
-      $('ul').html(items)
-      addListFilter()
+      resList = resources
```
Comment (zautumnz) [inline script.js:142]: These two lines could just be `fetchy(resources)`; `resList` is defined without a declaration, and creating the extra variable is useless in this context.

File: script.js [hunk 6]
```diff
@@ -58,62 +59,87 @@
...
 
   const buildResourceItem = ({ href, title, tags, description }) => {
     const link = `<a href="${href}" rel="noopener noreferrer nofollower" target="_blank">${title}</a/>`
     const tagList = `<small>tags: ${tags.join(' ')}</small>`
     return `<li>${[link, description, tagList].filter(Boolean).join('<br>')}</li>`
   }
 
+  const fetchy = (resources) => {
+    // enrich resource with plurals and tag aliases from functions above
+    const enrichedResources = enrichResources(resList)
```
Comment (zautumnz) [inline script.js:75]: I believe this should be `enrichResources(resources)`; resList isn't defined in this scope except by the global declaration below (see that comment).

File: script.js [hunk 7]
```diff
@@ -58,62 +59,87 @@
...
+
+    const fuse = new window.Fuse(enrichResources(resList), fuseOptions)
```
Comment (zautumnz) [inline script.js:77]: No need to call `enrichResources` again, this can just use `enrichedResources`.

File: script.js [hunk 8]
```diff
@@ -58,62 +59,87 @@
...
+
+    // set up filtering code
+    const addListFilter = () => {
+      const ul = $(document.body).find('ul')
+
+      // actual filtering function
+      const filterRows = (e) => {
+        //console.log("I wanna show all of it!")
+  
+        var completeList = itemsList.concat(items)
```
Comment (zautumnz) [inline script.js:87]: `const` please

File: script.js [hunk 9]
```diff
@@ -58,62 +59,87 @@
...
+    // links are added to the end of the yaml file
+    const items = enrichedResources.map(buildResourceItem).reverse()
+
+    const itemsList = items.splice(0,15)
+
+    // append all the items and add the filter
+    $('ul').html(itemsList)
+    addListFilter()
+
+    window.onscroll = function(e) {
```
Comment (zautumnz) [inline script.js:127]:
```suggestion
    window.onscroll = (e) => {
```

File: script.js [hunk 10]
```diff
@@ -58,62 +59,87 @@
...
+      }
+
+      $('#filter').on('click', filterRows)
+    }
+
+    // build the elements from the resources, and flip it since new
+    // links are added to the end of the yaml file
+    const items = enrichedResources.map(buildResourceItem).reverse()
+
+    const itemsList = items.splice(0,15)
```
Comment (zautumnz) [inline script.js:121]:
```suggestion
    const itemsList = items.slice(0, 15)
```

`splice` mutates the original array, `slice` does not. These splices progressively delete more of the original array. Need to take into account the current index and use a `slice`.

File: script.js [hunk 11]
```diff
@@ -58,62 +59,87 @@
...
+
+    // append all the items and add the filter
+    $('ul').html(itemsList)
+    addListFilter()
+
+    window.onscroll = function(e) {
+      if ((window.innerHeight + window.scrollY) >= document.body.offsetHeight) {
+          itemsList.push(...items.splice(0,15))
```
Comment (zautumnz) [inline script.js:129]:
```suggestion
          itemsList.push(...items.slice(0, 15))
```

File: script.js [hunk 12]
```diff
@@ -58,62 +59,87 @@
...
+
+          $('ul').html(itemsList.concat(resList.splice(0,15)))
```
Comment (zautumnz) [inline script.js:131]:
```suggestion
          $('ul').html(itemsList.concat(resList.slice(0, 15)))
```

File: script.js [hunk 13]
```diff
@@ -58,62 +59,87 @@
...
+
+      // actual filtering function
+      const filterRows = (e) => {
+        //console.log("I wanna show all of it!")
+  
+        var completeList = itemsList.concat(items)
+        //console.log(completeList)
+
+        $('ul').html(completeList)
+        addListFilter()
```
Comment (zautumnz) [inline script.js:91]: Probably because this is a draft, but calling `addListFilter` inside `addListFilter` won't be good.

**Review by zautumnz — CHANGES_REQUESTED** [review]
Thanks for the PR! I have a couple of comments and obviously you have it marked as a draft. I really appreciate the contributions :)

File: script.js (hunk 2 above)
Comment (m-ox) [reply script.js]: absolutely agree

**Comment by zautumnz** [comment]
Thanks for this. I merged it manually, because I'm going to archive this project and wanted your contribution to land first.

**Pull request merged**